agent/util/expansions.go

```diff
@@ -47,3 +51,25 @@ func (e *DynamicExpansions) Get(key string) string {
 
 	return e.Expansions.Get(key)
 }
+
+// Redact marks the expansion key for redaction.
+func (e *DynamicExpansions) Redact(key string) {
+	e.redactMu.Lock()
+	defer e.redactMu.Unlock()
+
+	e.redact = append(e.redact, key)
+}
+
+// GetRedacted gets the expansions that should be redacted.
```

Comment:
nit: can we mention this is specifically those expansions that were redacted by expansions.update? Or maybe rename it? Other expansions are redacted using other mechanisms (like project var redaction).

Reply:
Right after this PR, I'm going to add the token generated by `github.generate_token` command to also redact it.

I'll add wording to hint at expansions.write / other project commands that generate secrets